### ISSUE TITLE
Bottom Bar internal navigation - Help

### ISSUE THREAD

**Issue by ArleyPereira** [raamcosta/compose-destinations#238]
Hi Rafael, How are you?

I'm managing to do very well using your library, but I'm stuck in an internal navigation to the items in the bottom bar.

My bottom bar is made up of 4 screens:
FeedScreen()
SearchScreen()
FriendsScreen()
ProfileScreen()

On the ProfileScreen() screen, I need to open another screen called EditProfileScreen(), but the navigation doesn't work

I mount my bottom bar in HomeScreen()
https://prnt.sc/v8ewWnaYB2_8

Bottom Bar
https://prnt.sc/V7QVb-ZdfZic

BottomBarItem - sealed class
https://prnt.sc/VeDJm3z8DOom

ProfileScreen() - Click event for EditProfileScreen()
https://prnt.sc/PF8htwlVuxN4

Final destination
https://prnt.sc/CBVe3MapX4Vk

I've tried several ways and I can't navigate from ProfileScreen() contained in BottomBar to EditProfileScreen()

**Comment by ArleyPereira** [comment]
Hello Rafa, this is my bottom bar implementation

BottomBar
https://prnt.sc/V7QVb-ZdfZic

I don't have any errors, simply navigation is not done from ProfileScreen() to EditProfileScreen()

ProfileScreen() - Clique no evento para EditProfileScreen()
![image](https://user-images.githubusercontent.com/47366406/189875972-425d68ab-99c0-4092-b9d5-add59c1279e1.png)

Destino final
![image](https://user-images.githubusercontent.com/47366406/189876049-10642dc9-246a-44f6-9721-249001909c8c.png)

**Comment by ArleyPereira** [comment]
Hello Rafa, can you solve it, but I would like to know your opinion about the following behavior.

I created two graphs nested to the root graph

```
@RootNavGraph(start = true)
@NavGraph
annotation class AuthGraph(
    val start: Boolean = false
)
```

```
@RootNavGraph
@NavGraph
annotation class MainGraph(
    val start: Boolean = false
)
```

BottomBar Itens

```
@MainGraph(start = true)
@Destination
@Composable
fun FeedScreen(navigator: DestinationsNavigator)
```

```
@MainGraph
@Destination
@Composable
fun SearchScreen(navigator: DestinationsNavigator)
```

```
@MainGraph
@Destination
@Composable
fun FriendsScreen(navigator: DestinationsNavigator)
```

```
@MainGraph
@Destination
@Composable
fun ProfileScreen(navigator: DestinationsNavigator)
```

On the ProfileScreen() screen I have a click event that takes the user to the EditProfileScreen() screen.

On the Profile Screen() screen I have a click event that takes the user to the Edit Profile Screen() screen, but when I click I have the error below.

`Navigation destination that matches request NavDeepLinkRequest{ uri=android-app://androidx.navigation/edit_profile_screen?text=Arley } cannot be found in the navigation graph NavGraph(0xd750ce95) route=main_graph startDestination={Destination(0x857ec080) route=feed_screen}`

My EditProfileScreen() does not implement any explicit graph and the error occurs, when I implemented the @MainGraph graph I was successful

```
@MainGraph
@Destination
@Composable
fun EditProfileScreen(text: String = "")
```

After explaining the behavior above, my question is whether all my screens must belong to an explicit graph?

**Comment by ArleyPereira** [comment]
I found the problem. It was in the Destinations NavHost configuration. After opening several issues here on github I will choose to use only the root graph. Now I have another problem with the bottom bar, because when I tap the login button I go to the app home where I build the bottom bar, but the login screen remains open

![image](https://user-images.githubusercontent.com/47366406/189918153-91a9eed0-9fc8-45f9-8505-66f867395283.png)


```
navigator.navigate(HomeScreenDestination()) {
    popUpTo(NavGraphs.root.route) { inclusive = true }
}

navigator.navigate(HomeScreenDestination()) {
    popUpTo(LoginScreenDestination.route) { inclusive = true }
}
```
I've tried several ways to close the login screen and I couldn't. I also noticed that my BottomBar item 0 is not automatically selected
